### PR TITLE
gh-95855: Better platform triplet detection code

### PR DESCRIPTION
**Pull request by tiran**
* Issue: gh-95855

### CODE REVIEW

**Review by cougargt — APPROVED** [review]

**Comment by arhadthedev** [comment]
@tiran It would be great to see it merged.

File: Misc/platform_triplet.c [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+#  define LIBC_ARM_EABI gnueabi
+#  define LIBC_PPC_SPE gnuspe
+#  if defined(_MIPS_SIM)
+#   if _MIPS_SIM == _ABIO32
+        define LIBC_MIPS_SIM gnu
+#   elif _MIPS_SIM == _ABIN32
+        define LIBC_MIPS_SIM gnuabin32
+#   elif _MIPS_SIM == _ABI64
+        define LIBC_MIPS_SIM gnuabi64
+#   endif
```
Comment (jefferyto) [inline Misc/platform_triplet.c:40]:
```suggestion
#   if _MIPS_SIM == _ABIO32
#    define LIBC_MIPS_SIM gnu
#   elif _MIPS_SIM == _ABIN32
#    define LIBC_MIPS_SIM gnuabin32
#   elif _MIPS_SIM == _ABI64
#    define LIBC_MIPS_SIM gnuabi64
#   endif
```

**Comment by jefferyto** [comment]
I'm working on fixing/improving the platform triplet for Python on OpenWrt, and I think it would be better to build on this rather than the existing code in configure.ac.

**Comment by jefferyto** [comment]
@tiran is there anything blocking this PR? I have some updates to this change (in https://github.com/openwrt/packages/pull/21552, specifically in [this commit](https://github.com/openwrt/packages/pull/21552/commits/4ee56d006abd78f880fd662ebfd32b70d394b65e)) that I'd like to submit upstream.

**Comment by erlend-aasland** [comment]
Superseded by #107221

**Pull request closed without merging**